module/os/linux/zfs/abd_os.c

```diff
@@ -680,7 +685,13 @@ abd_free_zero_scatter(void)
 	abd_free_sg_table(abd_zero_scatter);
 	abd_free_struct(abd_zero_scatter);
 	abd_zero_scatter = NULL;
-#if !defined(_KERNEL)
+	ASSERT3P(abd_zero_page, !=, NULL);
+#if defined(_KERNEL)
+	abd_unmark_zfs_page(abd_zero_page);
+	int order = 0;
+	order = compound_order(abd_zero_page);
+	__free_pages(abd_zero_page, order);
```

Comment:
Why not use `__free_page(abd_zero_page);` here?  We know the order must be zero due to how it was allocated above. 

```c
 	abd_unmark_zfs_page(abd_zero_page);
 	__free_page(abd_zero_page);
```

Reply:
Good point, I will update this.